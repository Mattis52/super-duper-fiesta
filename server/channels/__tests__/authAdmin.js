jest.mock('../../models/meeting');
jest.mock('../../utils');
const { emit, broadcast } = require('../../utils');
const { createGenfors } = require('../../models/meeting');
const { generateGenfors, generateSocket } = require('../../utils/generateTestData');
const { createGenfors: createGenforsListener } = require('../auth');

const generateData = data => Object.assign({}, data);

describe('admin', () => {
  it('returns invalid admin password if incorrect', async () => {
    process.env.SDF_GENFORS_ADMIN_PASSWORD = 'correct';

    await createGenforsListener(generateSocket(), generateData());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);

    process.env.SDF_GENFORS_ADMIN_PASSWORD = '';
  });

  it('creates a new meeting if admin password is correct', async () => {
    createGenfors.mockImplementation(async () => generateGenfors());
    process.env.SDF_GENFORS_ADMIN_PASSWORD = 'correct';

    await createGenforsListener(generateSocket(), generateData({ password: 'correct' }));

    expect(emit.mock.calls).toEqual([]);
    expect(broadcast.mock.calls).toEqual([]);

    process.env.SDF_GENFORS_ADMIN_PASSWORD = '';
  });
});
