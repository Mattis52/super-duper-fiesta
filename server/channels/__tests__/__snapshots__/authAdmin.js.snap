// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin login returns invalid admin password if incorrect 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "AUTH_ERROR",
    Object {
      "error": "Ugyldig administratorpassord.",
    },
  ],
]
`;
